doc/sphinx-guides/source/developers/dataset-semantic-metadata-api.rst

```diff
@@ -0,0 +1,91 @@
+You may not want to use this API if you need stability and backward compatibility (the 'experimental' designation for this API implies that community feedback is desired and that, in future Dataverse software versions, the API may be modified based on that feedback).
+
+Note: The examples use the 'application/ld+json' mimetype. For compatibility reasons, the APIs also be used with mimetype "application/json-ld"
+  
+Get Dataset Metadata
+--------------------
+
+To get the json-ld formatted metadata for a Dataset, specify the Dataset ID (DATASET_ID) or Persistent identifier (DATASET_PID), and, for specific versions, the version number.
+
+.. code-block:: bash
+
+  export API_TOKEN=xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+  export DATASET_ID='12345'
+  export DATASET_PID='doi:10.5072/FK2A1B2C3'
+  export VERSION='1.0'
+  export SERVER_URL=https://demo.dataverse.org
+ 
+  Example 1: Get metadata for version '1.0'
+ 
+    curl -H X-Dataverse-key:$API_TOKEN -H 'Accept: application/ld+json' "$SERVER_URL/api/datasets/$DATASET_ID/versions/$VERSION/metadata"
+
+  Example 2: Get metadata for the latest version using the DATASET PID
+
+    curl -H X-Dataverse-key:$API_TOKEN -H 'Accept: application/ld+json' "$SERVER_URL/api/datasets/:persistentId/metadata?persistentId=$DATASET_PID"
+
+You should expect a 200 ("OK") response and JSON-LD mirroring the OAI-ORE representation in the returned 'data' object.
+
+
+Add Dataset Metadata
+--------------------
+
+To add json-ld formatted metadata for a Dataset, specify the Dataset ID (DATASET_ID) or Persistent identifier (DATASET_PID). Adding '?replace=true' will overwrite an existing metadata value. The default (replace=false) will only add new metadata or add a new value to a multi-valued field. 
+
+.. code-block:: bash
+
+  export API_TOKEN=xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+  export DATASET_ID='12345'
+  export DATASET_PID='doi:10.5072/FK2A1B2C3'
+  export VERSION='1.0'
+  export SERVER_URL=https://demo.dataverse.org
+ 
+  Example: Change the Dataset title 
+ 
+    curl -X PUT -H X-Dataverse-key:$API_TOKEN -H 'Content-Type: application/ld+json' -d '{"Title": "Submit menu test", "@context":{"Title": "http://purl.org/dc/terms/title"}}' "$SERVER_URL/api/datasets/$DATASET_ID/metadata?replace=true"
+
+  Example 2: Add a description using the DATASET PID
+
+    curl -X PUT -H X-Dataverse-key:$API_TOKEN -H 'Content-Type: application/ld+json' -d '{"citation:Description": {"dsDescription:Text": "New description"}, "@context":{"citation": "https://dataverse.org/schema/citation/","dsDescription": "https://dataverse.org/schema/citation/dsDescription#"}}' "$SERVER_URL/api/datasets/:persistentId/metadata?persistentId=$DATASET_PID"
```

Comment:
Should we have URLs like https://dataverse.org/schema/citation/dsDescription actually work?

Reply:
That would be useful. These URLs were defined a while back with the changes to add URLs to the metadata blocks and export them in the OAI-ORE/Bag structures. It would probably require some discussion of what should actually be available from such links and where that info would come from (given that new metadata blocks can define new URLs).

Reply:
It would be great to have some review of these 'internal terms' both to see if there are external terms that make sense (I did try to avoid them when I did the original archiving work but some terms either didn't have a match at all or matched in meaning but didn't allow the values Dataverse has (either multiple versus single values or things like contact having to be a person or requiring a phone number in the value when Dataverse doesn't have that, etc. - basically term range issues). Doing that and doing something more formal to give the remaining real ones live URLs with some info about that term (deciding whether the definitions are community wide (with a single community URL) or are instance specific (with URLs related to the specific machine), etc.) would be good. Perhaps at least creating an issue would be a good step?